### PR TITLE
[DOC] Improve doc guide compliance

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Avoid suppressing autolinks to classes and modules.

I think this is the last for ruby/*.{c,rb}.  Everything else is upstream, right?

### CODE REVIEW

File: object.c [hunk 1]
```diff
@@ -4298,7 +4298,7 @@ InitVM_Object(void)
      *
      * == What's Here
      *
-     * \Module \Kernel provides methods that are useful for:
```
Comment (peterzhu2118) [inline object.c:4301]: This is documentation for the `Kernel` module, so we don't need to unescape it.

File: re.c [hunk 2]
```diff
@@ -3498,7 +3498,7 @@ match_hash(VALUE match)
  *  call-seq:
  *    matchdata == object -> true or false
  *
- *  Returns +true+ if +object+ is another \MatchData object
```
Comment (peterzhu2118) [inline re.c:3501]: This is documentation for the `MatchData#==` right?

File: struct.c [hunk 3]
```diff
@@ -1547,7 +1547,7 @@ rb_struct_dig(int argc, VALUE *argv, VALUE self)
 /*
  *  Document-class: Data
  *
- *  \Class \Data provides a convenient way to define simple classes
+ *  \Class Data provides a convenient way to define simple classes
```
Comment (peterzhu2118) [inline struct.c:1550]: This is documentation for the `Data` class, so we should escape it.

File: struct.c [hunk 4]
```diff
@@ -1619,7 +1619,7 @@ rb_struct_dig(int argc, VALUE *argv, VALUE self)
  * call-seq:
  *   define(*symbols) -> class
  *
- *  Defines a new \Data class.
+ *  Defines a new Data class.
```
Comment (peterzhu2118) [inline struct.c:1622]: Ditto.

File: struct.c [hunk 5]
```diff
@@ -1634,7 +1634,7 @@ rb_struct_dig(int argc, VALUE *argv, VALUE self)
  *     #=> #<data Measure amount=1, unit="km">
  *
  *
- *  Note that member-less \Data is acceptable and might be a useful technique
+ *  Note that member-less Data is acceptable and might be a useful technique
```
Comment (peterzhu2118) [inline struct.c:1637]: Ditto.

File: object.c (hunk 1 above)
Comment (BurdetteLamar) [reply object.c]: Fixed.

File: struct.c (hunk 4 above)
Comment (BurdetteLamar) [reply struct.c]: Fixed.

File: struct.c (hunk 5 above)
Comment (BurdetteLamar) [reply struct.c]: Fixed.

**Comment by BurdetteLamar** [comment]
@peterzhu2118, why are we having more than one class/module defined in a file?  It makes this operation (and perhaps others) more difficult.

**Comment by peterzhu2118** [comment]
Some classes need to share internal implementation in the C code, so they live in the same file. Unfortunately we can't resolve this issue without some (possibly significant) refactors in the C code.

File: object.c [hunk 6]
```diff
@@ -4143,9 +4143,9 @@ f_sprintf(int c, const VALUE *v, VALUE _)
  *
  *  === What's Here
  *
- *  These are the methods defined for \BasicObject:
+ *  These are the methods defined for BasicObject:
  *
- *  - ::new: Returns a new \BasicObject instance.
+ *  - ::new: Returns a new BasicObject instance.
```
Comment (peterzhu2118) [inline object.c:4148]: This is documentation for the BasicObject class.

Comment (BurdetteLamar) [reply object.c]: Fixed.

**Review by peterzhu2118 — APPROVED** [review]

**Pull request merged**